physics/Disk/include/accretion.hpp

```diff
@@ -0,0 +1,77 @@
+//
+// Created by Noah Kubli on 15.03.2024.
+//
+
+#pragma once
+
+#include <array>
+#include <cstdio>
+#include <mpi.h>
+
+#include "accretion_impl.hpp"
+#include "accretion_gpu.hpp"
+#include "cstone/primitives/mpi_wrappers.hpp"
+#include "cstone/primitives/accel_switch.hpp"
+
+namespace disk
+{
+
+//! @brief Flag particles for removal. Overwrites keys for the removed particles.
+template<typename Dataset, typename StarData>
+void computeAccretionCondition(size_t first, size_t last, Dataset& d, StarData& star)
+{
+    if constexpr (cstone::HaveGpu<typename Dataset::AcceleratorType>{})
+    {
+        computeAccretionConditionGPU(first, last, d, star);
+    }
+    else { computeAccretionConditionImpl(first, last, d, star); }
+}
+
+//! @brief Exchange accreted mass and momentum between ranks and add to star.
+template<typename StarData>
+void exchangeAndAccreteOnStar(StarData& star, double minDt_m1, int rank)
+{
+    double                m_accreted_global{};
+    std::array<double, 3> p_accreted_global{};
+
+    double                m_removed_global{};
+    std::array<double, 3> p_removed_global{};
+
+    MPI_Reduce(&star.accreted_local.mass, &m_accreted_global, 1, MpiType<double>{}, MPI_SUM, 0, MPI_COMM_WORLD);
+    MPI_Reduce(star.accreted_local.momentum.data(), p_accreted_global.data(), 3, MpiType<double>{}, MPI_SUM, 0,
+               MPI_COMM_WORLD);
+    MPI_Reduce(&star.removed_local.mass, &m_removed_global, 1, MpiType<double>{}, MPI_SUM, 0, MPI_COMM_WORLD);
+    MPI_Reduce(star.removed_local.momentum.data(), p_removed_global.data(), 3, MpiType<double>{}, MPI_SUM, 0,
```

Comment:
These `MPI_Reduces` can/should be combined into a single `MPI_Reduce` for better performance. Each call is latency bound, the combined call would be 4 times faster.

Reply:
This could be turned into a single `allreduce` and following computation be done on all ranks

Reply:
The data is now copied to a buffer with one single MPI_Allreduce call.